core/services/remote_access_services_test.py

```diff
@@ -0,0 +1,103 @@
+# coding: utf-8
+#
+# Copyright 2017 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for remote access services."""
+
+import json
+
+from core.services import remote_access_services
+from core.tests import test_utils
+import vmconf
+
+class RemoteAccessServicesTests(test_utils.GenericTestBase):
+    """Tests for remote accessing methods."""
+
+    def test_that_generate_signature_works_correctly(self):
+        """Test that generate signature function is working as expected."""
+        data = {
+            'vm_id': 'vm_default'
+        }
+
+        with self.swap(vmconf, 'DEFAULT_VM_SHARED_SECRET', '1a2b3c4e'):
+            sign = remote_access_services.generate_signature(data)
```

Comment:
Yeah it is `1a2b3c4e` as of now but if we decide to change that in future then this test will fail. Therefore I am swapping it with the value using which I have calculated `expected_sign` so test won't fail even if we change default key (since the purpose is to verify that `generate_signature()` is working and should not affect if we change default key).